### PR TITLE
fix(insights): map HogQL left()/right() to leftUTF8()/rightUTF8()

### PR DESCRIPTION
**Pull request by robbie-c**
## Problem

Error tracking issue: https://us.posthog.com/project/2/error_tracking/019ceb15-7b38-7263-9683-4722c1cf0d59

Users with non-ASCII breakdown values (e.g. Cyrillic, CJK) hit a `TypeError: sequence item 0: expected str instance, bytes found` in trends queries when the breakdown value exceeds `BREAKDOWN_VALUE_MAX_LENGTH` (400).

**Root cause:** ClickHouse's `left()` function operates on **bytes**, not UTF-8 characters. When the 400-byte truncation lands in the middle of a multi-byte character (e.g. Cyrillic chars are 2 bytes each), the result is invalid UTF-8. The `clickhouse-driver` can't decode invalid UTF-8 and returns Python `bytes` instead of `str`, which then causes a `TypeError` when `_format_breakdown_label` tries to `"::".join(labels)`.

63 occurrences today across 9 users, active since 2026-03-14.

## Changes

FYI - I made a potentially controversial change to the hogql function mapping here, I remapped left to leftUTF8 rather than adding leftUTF8 as its own thing.

I did this because I'm worried about leaving this footgun lying around, but it's different behaviour to what we've done for leftPad. Ultimately I'm easy either way as long as we can fix this query.

AI PR summary continues:

- Map HogQL's `left()` → ClickHouse's `leftUTF8()` and `right()` → `rightUTF8()` in the function registry. This is correct since HogQL should always handle strings as UTF-8 text, not raw bytes.
- Add regression test that creates a 401-char Cyrillic string (802 bytes) to trigger truncation at the character boundary.
- Update all affected snapshots.

## How did you test this code?

I'm an agent — no manual testing was performed. Automated testing:

- Verified `left('text', N)` returns `bytes` when truncation splits a multi-byte UTF-8 character, while `leftUTF8()` correctly returns `str`
- Regression test `test_trends_breakdown_with_multibyte_value_at_truncation_boundary` fails without the fix, passes with it
- All existing snapshot tests pass with updated snapshots

## Publish to changelog?

No

## Docs update

N/A

## 🤖 LLM context

Authored by Claude Code (Opus 4.6). The key debugging insight was that ClickHouse's `left()` is byte-based — confirmed by running `SELECT left('Нам', 3)` which returns `b'\xd0\x9d\xd0'` (3 bytes, invalid UTF-8) vs `leftUTF8('Нам', 3)` which returns `'Нам'` (3 characters).

### CODE REVIEW

File: posthog/hogql_queries/insights/trends/test/test_trends_query_runner.py [hunk 1]
```diff
@@ -7204,6 +7204,49 @@ def test_group_node_property_filter_types(self, _name, property_filter, expected
...
+            ),
+        )
+
+        assert len(response.results) == 1
+        label = response.results[0]["label"]
+        assert isinstance(label, str)
+        # Verify truncation happened at character boundary, not byte boundary
+        assert len(label) == 400
+        assert label == "Б" * 400
+        assert response.results[0]["count"] == 1
```
[the remark this answers is not in the document]
Comment (robbie-c) [reply posthog/hogql_queries/insights/trends/test/test_trends_query_runner.py:7248]: this doesn't add anything

**Review by mariusandra — APPROVED** [review]
This is probably fine... though I'm curious about what might happen to existing queries. Hopefully nothing substantial.
